src/eva/vision/data/datasets/classification/camelyon16.py

```diff
@@ -0,0 +1,224 @@
+        "normal_019",
+        "normal_020",
+        "normal_025",
+        "normal_030",
+        "normal_031",
+        "normal_032",
+        "normal_052",
+        "normal_056",
+        "normal_057",
+        "normal_067",
+        "normal_076",
+        "normal_079",
+        "normal_085",
+        "normal_095",
+        "normal_098",
+        "normal_099",
+        "normal_101",
+        "normal_102",
+        "normal_105",
+        "normal_106",
+        "normal_109",
+        "normal_129",
+        "normal_132",
+        "normal_137",
+        "normal_142",
+        "normal_143",
+        "normal_148",
+        "normal_152",
+        "tumor_001",
+        "tumor_005",
+        "tumor_011",
+        "tumor_012",
+        "tumor_013",
+        "tumor_019",
+        "tumor_031",
+        "tumor_037",
+        "tumor_043",
+        "tumor_046",
+        "tumor_057",
+        "tumor_065",
+        "tumor_069",
+        "tumor_071",
+        "tumor_073",
+        "tumor_079",
+        "tumor_080",
+        "tumor_081",
+        "tumor_082",
+        "tumor_085",
+        "tumor_097",
+        "tumor_109",
+    ]
+    """Validation slide names, same as the ones in patch camelyon."""
+
+    def __init__(
+        self,
+        root: str,
+        sampler: samplers.Sampler,
+        split: Literal["train", "val", "test"] | None = None,
+        width: int = 224,
+        height: int = 224,
+        target_mpp: float = 0.5,
+        backend: str = "openslide",
+        image_transforms: Callable | None = None,
+        seed: int = 42,
+    ) -> None:
+        """Initializes the dataset.
+
+        Args:
+            root: Root directory of the dataset.
+            sampler: The sampler to use for sampling patch coordinates.
+            split: Dataset split to use. If `None`, the entire dataset is used.
+            width: Width of the patches to be extracted, in pixels.
+            height: Height of the patches to be extracted, in pixels.
+            target_mpp: Target microns per pixel (mpp) for the patches.
+            backend: The backend to use for reading the whole-slide images.
+            image_transforms: Transforms to apply to the extracted image patches.
+            seed: Random seed for reproducibility.
+        """
+        self._split = split
+        self._root = root
+        self._width = width
+        self._height = height
+        self._target_mpp = target_mpp
+        self._seed = seed
+
+        wsi.MultiWsiDataset.__init__(
+            self,
+            root=root,
+            file_paths=self._load_file_paths(split),
+            width=width,
+            height=height,
+            sampler=sampler,
+            target_mpp=target_mpp,
+            backend=backend,
+            image_transforms=image_transforms,
+        )
+
+    @property
+    @override
+    def classes(self) -> List[str]:
+        return ["0", "1"]
+
+    @functools.cached_property
+    def test_annotations(self) -> Dict[str, str]:
```

Comment:
Maybe we could rename this to `annotations` and store the annotations for train / val here as well?
Could use `self. file_paths` & `_get_target_from_path` for train/val, and the `testing/reference.csv` for loading the test labels.

Reply:
i wouldn't do this here, because `testing/reference.csv` is a raw metadata file. Better to use it original and as intended than to modify (and confuse users).

Reply:
I'm not suggesting to modify the `testing/reference.csv` file, but to have a `annotations` property that contains all annotations, and not only the ones of the test set. 

Two me this has two main advantages:
1. Cleaner code structure
2. Simplifies the `_get_target_from_path` function to:
```python
def _get_target_from_path(self, file_path: str) -> int:
    return self.annotations_test_set[self._get_id_from_path(file_path)]
```

Probably then we don't even need a seperate function for `_get_target_from_path`, but we could do that directly in the `load_target` function.

Reply:
refactored this -- packed the logic into `annotations` and got rid off `_get_target_from_path`. Let me know if this is how you think it should look like?

Reply:
amazing! thanks for the refactoring